src/Symfony/Component/Yaml/Inline.php

```diff
@@ -562,6 +562,8 @@ private static function evaluateScalar($scalar, $flags, $references = array())
                         }
 
                         return;
+                    case 0 === strpos($scalar, '!php/const:'):
+                        return @constant(substr($scalar, 11));
```

Comment:
I'm almost there :)

Since the constant are not autoloaded (IIUC), one may expect to get `null` if a constant is not defined in an environment, so I guess we can take advantage of the triggered `E_WARNING` (ref http://php.net/manual/en/function.constant.php) as it will be shown as a silenced error in the profiler thanks to https://github.com/symfony/symfony/pull/16760.

Hence, I suggest we only trigger an exception when `PARSE_EXCEPTION_ON_INVALID_TYPE` flag is passed.

I already did such implementation but I've got an issue when the constant is not "inlined":
```yaml
foo:
    - '!php/const:PHP_INT_MAX' # does not work
bar: ['!php/const:PHP_INT_MAX'] # works
baz: { foo: '!php/const:PHP_INT_MAX' } # works
```

I think it comes from the `Parser` as I only changed the `Inline` class for now, any lead ?

Thanks.